includes/validation/class-amp-validation-error-taxonomy.php

```diff
@@ -439,10 +446,12 @@ public static function summarize_validation_errors( $validation_errors ) {
 			}
 
 			if ( ! empty( $validation_error['sources'] ) ) {
-				$source = array_shift( $validation_error['sources'] );
+				$sources = array_shift( $validation_error['sources'] );
 
-				if ( isset( $source['type'], $source['name'] ) ) {
-					$invalid_sources[ $source['type'] ][] = $source['name'];
+				foreach ( $sources['sources'] as $source ) {
```

Comment:
^ Changed this since the sources were not properly displaying in the table -- the `type` and `name` were never set since the actual structure of `$validation_error` seems to be like this:
![screen shot 2018-09-04 at 5 21 41 pm](https://user-images.githubusercontent.com/3294597/45038305-a6207600-b069-11e8-89ad-78f3fefea0f3.png)

Meaning that the `type` and `name` would be set in `$validation_error['sources']['sources'][0]`, not in `$validation_error['sources']`.

However, this change broke the tests since the input data of test is assuming that a validation error looks like this instead:
![screen shot 2018-09-04 at 5 43 21 pm](https://user-images.githubusercontent.com/3294597/45038540-13340b80-b06a-11e8-9b30-a83b531504b7.png)

Is the structure of validation error incorrect for some reason and it should be as assumed in the tests? Maybe the fix should be removing the second nested `sources` instead, or is it like this for some reason? Thoughts?

cc @kienstra @westonruter

Reply:
Hi @miina,
That's interesting. I couldn't reproduce that issue.

It looks like there aren't double-nested `'sources'` arrays, like in your first screenshot when I run the `develop` branch:

<img width="977" alt="name-type-local" src="https://user-images.githubusercontent.com/4063887/45173577-3d9ed980-b1ce-11e8-865f-308dcbfa4c13.png">

Maybe the validation error data model changed.

It might help to delete all validation error posts and terms, and regenerate them (thanks to Weston's code, though I might not have copied it right):

```bash
wp post delete $(wp post list --post_type='amp_invalid_url' --format=ids)
wp term list amp_validation_error --field=term_id | xargs wp term delete amp_validation_error
```
You might already have some logic to output validation errors, but here's a [simple plugin](https://gist.github.com/kienstra/c204c13a0974d949ed741f37f8f9e14f) I've used (also partially copied from Weston)

And the WP-CLI script can then regenerate errors:
```bash 
cd /path/in/vvv/wp-content/plugins/amp
wp amp validate-site limit=5
```